RNS Shell minor improvments

munezbn opened the pull request:
- eglGetProcAddress() is not guaranteed to support the querying of non-extension EGL functions so improved egl_get() function
- Added two internal API getMainWindow() and getBufferAge() to get Apps main window and age of current opengl backbuffer
- Added new interface to get bufferage from compositor.
- Defination of GLNativeWindowType was in multiple headers, Now we will have at only one place(header).
- Fixed typo in EGL_EGLEXT_PROTOTYPES macro
- Added initialization for eglSetDamageRegion if supported by the platform.
- Optimized addDamageRect function.
  - If new dirty rect fully covers any of existing dirtyRect in the vector, then remove them from vector
  - If new dirty rect is inside any of existing dirtyRects, then ignore and return
- Added additional debug logs.

munezbn committed:
RNS Shell minor improvments

eglGetProcAddress() is not guaranteed to support the querying of non-extension EGL functions so improved egl_get() function
Added two internal API getMainWindow() and getBufferAge() to get Apps main window and age of current opengl backbuffer
Defination of GLNativeWindowType was in multiple headers, Now we will have at only one place(header).
Added aditional debug logs.

munezbn committed:
Updated Default value for swap behaviour query.

munezbn committed:
Updated Defeault value for swap behaviour query.

munezbn committed:
Some more minor improvments and fixes in rns_shell

Added new interface to get bufferage from compositor.
Fixed typo in EGL_EGLEXT_PROTOTYPES macro
Added initilization for eglSetDamageRegion if supported by the platform.

munezbn committed:
Merge branch 'munez/wip/graphics' of https://github.com/nagra-opentv/react-native-skia into munez/wip/graphics

pprabhu-ng commented on rns_shell/compositor/Compositor.h:
```diff
@@ -61,7 +61,7 @@ class Compositor {
     std::unique_ptr<WindowContext> windowContext_;
     sk_sp<SkSurface> backBuffer_;
-    uint64_t nativeWindowHandle_;
+    GLNativeWindowType nativeWindowHandle_;
```
in createWindowContext() we may not require the typecast on nativeWindowHandle_ now ?

munezbn replied:
Done

shefayaparwin commented on rns_shell/platform/graphics/gl/GLWindowContext.cpp:
```diff
@@ -93,6 +93,11 @@ bool GLWindowContext::hasSwapBuffersWithDamage() {
 }
+
+int32_t GLWindowContext::bufferAge() {
+    return this->getBufferAge();
```
For non Gpu boxes, is this implementation applicable ?

munezbn replied:
Fixed now. bufferAge() pure virtual function will be added only for GPU configuration now.

munezbn committed:
Addressed PR review comments

BUfferAge must not be used for non-GPU configuration.
Fixed typecast

munezbn committed:
Optimized addDamageRect function.

If new dirty rect fully covers any of existing dirtyRect in the vector, then remove them from vector
If new dirty rect is inside any of existing dirtyRects, then ignore and return
